controllers/topology/desired_state.go

```diff
@@ -170,6 +182,92 @@ func computeCluster(current *clusterTopologyState, infrastructureCluster, contro
 	return cluster
 }
 
+// computeMachineDeployment computes the desired state for a MachineDeploymentTopology.
+// The generated machineDeployment object is calculated using the values from the machineDeploymentTopology and
+// the machineDeployment class.
+func computeMachineDeployment(class *clusterTopologyClass, current *clusterTopologyState, machineDeploymentTopology clusterv1.MachineDeploymentTopology) (machineDeploymentTopologyState, error) {
+	desiredMachineDeployment := machineDeploymentTopologyState{}
+
+	className := machineDeploymentTopology.Class
+	machineDeploymentTopologyClass, ok := class.machineDeployments[className]
```

Comment:
```suggestion
	machineDeploymentClass, ok := class.machineDeployments[className]
```
Let use the `class` suffix for everything that comes from the `ClusterClass` and the `topology` suffix for everything that comes from `Cluster.Topology` (but not mix them)